### ISSUE TITLE
regression: unvanquished messes up with textures from models baked in maps

### ISSUE THREAD

**Issue by bmorel** [Unvanquished/Unvanquished#2799]
![2023-10-04](https://github.com/Unvanquished/Unvanquished/assets/1316300/bc15ca2d-1d63-4745-9a94-dda3acd73f9c)

While I do not know what this should look like, the behavior is very close to what can be seen in @cu-kai 's station12-b5 map, which should look like this:

![2023-10-04_000](https://github.com/Unvanquished/Unvanquished/assets/1316300/4b6b8e1d-e853-4a3f-a9d5-18b275285e98)

but really looks like that:

![2023-10-04_001](https://github.com/Unvanquished/Unvanquished/assets/1316300/6199d730-1d13-4b0e-ba64-ff746e2c5841)

I could hackily "fix" that map ported by @sweet235 by modifying the bsp with an hexadecimal editor. Can you see the change in this screenshot? You should, it's rather obvious.

![2023-10-04_002](https://github.com/Unvanquished/Unvanquished/assets/1316300/03c5d80c-21c8-43e9-a6dd-b93de6d3d139)

I also renamed files in the dpk:

```
map-station12-b5_src.dpkdir% find -iname '*level9*'
./models/players/level9
./models/players/level9/level9adv.tga
./models/players/level9/level9.tga
```

and modified "random" files:

```
map-station12-b5_src.dpkdir% grep -r level9
models/players/level9/nonseg_default.skin:level9,models/players/level9/level9.tga
models/players/level9/nonseg_adv.skin:level9,models/players/level9/level9adv.tga
models/players/level9/animation.cfg:// Tremulous: level9 alien
```

The problem appears on other maps, such as AMP's "example" map where flame files are "missing" (not in the correct place, really) and probably several other ones that I don't remember right now.
Any map using baked-in model of stuff from tremulous or textures from tremulous is likely to be affected in various unpredictable ways.

**Comment by slipher** [comment]
A `models/players/level3/level3` shader is defined for our Unvanquished dragoon model. The basepak is loaded first and the map last, so the Unvanquished one wins. As it should be: the alternative bug of having the map override Unvanquished assets would be much worse!

**Comment by bmorel** [comment]
> As it should be

Not really, as models are baked into bsp files and can't be modified without resorting to ugly hacks like I did.

> the alternative bug of having the map override Unvanquished assets would be much worse!

I would argue that the problem comes from unvanquished using the same names for models, in addition of models not embedding their textures.
If unvanquished used a "namespace" or a prefix (let's say, randomly, "unv_" then there would not be any collision.

**Comment by sweet235** [comment]
I don't get it. Why is only the dragoon texture flipped/rotated, but not the other ones?

**Comment by slipher** [comment]
> I don't get it. Why is only the dragoon texture flipped/rotated, but not the other ones?

It depends on whether there are name conflicts with Unvanquished assets

**Comment by illwieckz** [comment]
This is because Tremulous name model textures like this:

- `models/level0/level0`
- `models/mgturret/mgturret`

And Unvanquished name model textures like this:

- `models/level0/level0`
- `models/mgturret/mgturret`

It would have been better if that was done instead:

- `models/tremulous/level0/level0`
- `models/tremulous/mgturret/mgturret`
- `models/unvanquished/level0/level0`
- `models/unvanquished/mgturret/mgturret`

The `res-tremulous` package moved Tremulous conflicting model textures to `model/tremulous` to avoid breaking the game models: choice was made to break legacy Tremulous model textures instead of letting legacy Tremulous maps break the game.

I would like to move specific unvanquished things to `models/unvanquished` and I may have a branch for that but that's a lot of work.

**Comment by illwieckz** [comment]
The alternative was that every time you loaded a legacy Tremulous map relying on `res-tremulous` packages you would have stock Unvanquished players, buildables and weapons models as broken as this.

**Comment by illwieckz** [comment]
> I don't get it. Why is only the dragoon texture flipped/rotated, but not the other ones?

It happens that sometime, new models/textures are using some other name for various reasons, for example the previous lucifer canon was named _pulse_ because it was initially modeled to be a pulse rifle, and tremulous never had `models/weapons/lcannon/pulse` texture so there could have been no conflict.

**Comment by DolceTriade** [comment]
Should we just rename the unv textures to something else so they don't conflict with legacy textures?

**Comment by illwieckz** [comment]
Yes but it's better if we do this the reliable way, like adding a prefix folder.

Adding a prefix folder is a kind of renaming, but with an agreed convention.
